x-pack/legacy/plugins/ml/public/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/reducer.ts

```diff
@@ -61,7 +61,7 @@ export const validateAdvancedEditor = (state: State): State => {
   const destinationIndexName = idx(jobConfig, _ => _.dest.index) || '';
   const destinationIndexNameEmpty = destinationIndexName === '';
   const destinationIndexNameValid = isValidIndexName(destinationIndexName);
-  const destinationIndexPatternTitleExists = state.indexPatternTitles.some(
+  const destinationIndexPatternTitleExists = Object.keys(state.indexPatternsMap).some(
```

Comment:
could this be `destinationIndexPatternTitleExists = state.indexPatternsMap[destinationIndexName] !== undefined`?
There are a few places where the keys of `indexPatternsMap` are looped over, which make me wonder if using an object for storing this data is the best choice.
I agree with a previous suggestion that perhaps a `Map` could be used here.